usl_models/usl_models/flood_ml/prediction_dataset.py

```diff
@@ -0,0 +1,107 @@
+"""tf.data.Datasets for prediction using FloodML model trained on CityCAT data."""
+
+import logging
+from typing import Dict, Iterator, Optional, TypedDict
+
+from google.cloud import storage  # type:ignore[attr-defined]
+import numpy
+import tensorflow as tf
+
+from usl_models.flood_ml import constants, dataset
+
+
+class PredictionInput(TypedDict):
+    """Input tensors dictionary used for prediction."""
+
+    geospatial: tf.Tensor
+    temporal: tf.Tensor
+    spatiotemporal: tf.Tensor
+    metadata: tf.Tensor
+
+
+def load_prediction_dataset(
+    rainfall_config_gcs_uri: str,
+    chunk_id_to_gcs_uri_map: Dict[str, str],
```

Comment:
I think the interface we want is what @prashantkul  implemented over here:

https://github.com/UrbanSystemsLab/climateiq-cnn/pull/145/files#diff-d2183c123f3657caeb8ad0386b9f4e6aad151f39f95fcc0c2dbaf8d22678d705R21

where the caller provides the study area and config name, and the function itself is responsible for doing metastore lookups.